src/main/java/org/reactome/server/service/controller/citation/CitationController.java

```diff
@@ -0,0 +1,294 @@
+package org.reactome.server.service.controller.citation;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.annotations.ApiParam;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.reactome.server.graph.domain.model.DatabaseObject;
+import org.reactome.server.graph.domain.model.InstanceEdit;
+import org.reactome.server.graph.domain.model.Pathway;
+import org.reactome.server.graph.domain.model.Person;
+import org.reactome.server.graph.service.AdvancedDatabaseObjectService;
+import org.reactome.server.graph.service.GeneralService;
+import org.reactome.server.service.model.citation.Citation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.w3c.dom.Document;
+import org.xml.sax.InputSource;
+import springfox.documentation.annotations.ApiIgnore;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.*;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Yusra Haider (yhaider@ebi.ac.uk)
+ * @since 11.02.2
+ */
+
+@ApiIgnore
+@RestController
+@RequestMapping("/citation")
+public class CitationController {
+
+    private static final Logger infoLogger = LoggerFactory.getLogger("infoLogger");
+    private static final String STATIC_CITATION_ERROR = "Unable to fetch static citation";
+    private static final String DOI_BASE_URL = "https://doi.org/";
+
+
+    @Autowired
+    private AdvancedDatabaseObjectService advancedDatabaseObjectService;
+    @Autowired
+    private GeneralService generalService;
```

Comment:
My IntelliJ doesn't like field annotation like`@Autowired`. It suggests to apply either in the constructor or on a setter.